components/econet/econet.cpp

```diff
@@ -541,13 +541,16 @@ void Econet::send_datapoint_(const std::string &datapoint_id, const EconetDatapo
   datapoints_[datapoint_id] = value;
   for (auto &listener : this->listeners_) {
     if (listener.datapoint_id == datapoint_id) {
-      listener.on_datapoint(value);
+      if (listener.src_adr == 0 || listener.src_adr == src_adr) {
```

Comment:
nit: combine this with previous line
`if (listener.datapoint_id == datapoint_id && (listener.src_adr == 0 || listener.src_adr == src_adr)) {`